### ISSUE TITLE
Spec errors under 1.9 alphas

### ISSUE THREAD

**Issue by jjl** [ztellman/aleph#279]
```
clojure.lang.Compiler$CompilerException: clojure.lang.ExceptionInfo: Call to clojure.core/defn did not conform to spec:
                                         In: [2 0] val: ({:keys [name content mime-type charset transfer-encoding], :or {:transfer-encoding :quoted-printable}}) fails spec: :clojure.core.specs/arg-list at: [:args :bs :arity-1 :args] predicate: (cat :args (* :clojure.core.specs/binding-form) :varargs (? (cat :amp #{(quote &)} :form :clojure.core.specs/binding-form))),  Extra input
                                         In: [2 0] val: {:keys [name content mime-type charset transfer-encoding], :or {:transfer-encoding :quoted-printable}} fails spec: :clojure.core.specs/arg-list at: [:args :bs :arity-n :bodies :args] predicate: vector?
                                         :clojure.spec/args  (populate-part "Generates a part map of the appropriate format" [{:keys [name content mime-type charset transfer-encoding], :or {:transfer-encoding :quoted-printable}}] (let [mt (or mime-type (when (instance? File content) (URLConnection/guessContentTypeFromName (.getName content))))] {:name name, :content (bs/to-byte-buffer content), :mime-type (mime-type-descriptor mt charset), :transfer-encoding transfer-encoding}))
                                          #:clojure.spec{:problems ({:path [:args :bs :arity-1 :args], :reason "Extra input", :pred (cat :args (* :clojure.core.specs/binding-form) :varargs (? (cat :amp #{(quote &)} :form :clojure.core.specs/binding-form))), :val ({:keys [name content mime-type charset transfer-encoding], :or {:transfer-encoding :quoted-printable}}), :via [:clojure.core.specs/defn-args :clojure.core.specs/args+body :clojure.core.specs/arg-list :clojure.core.specs/arg-list], :in [2 0]} {:path [:args :bs :arity-n :bodies :args], :pred vector?, :val {:keys [name content mime-type charset transfer-encoding], :or {:transfer-encoding :quoted-printable}}, :via [:clojure.core.specs/defn-args :clojure.core.specs/args+body :clojure.core.specs/args+body :clojure.core.specs/args+body :clojure.core.specs/arg-list :clojure.core.specs/arg-list], :in [2 0]}), :args (populate-part "Generates a part map of the appropriate format" [{:keys [name content mime-type charset transfer-encoding], :or {:transfer-encoding :quoted-printable}}] (let [mt (or mime-type (when (instance? File content) (URLConnection/guessContentTypeFromName (.getName content))))] {:name name, :content (bs/to-byte-buffer content), :mime-type (mime-type-descriptor mt charset), :transfer-encoding transfer-encoding}))}, compiling:(aleph/http/multipart.clj:30:1)
             clojure.lang.ExceptionInfo: Call to clojure.core/defn did not conform to spec:
                                         In: [2 0] val: ({:keys [name content mime-type charset transfer-encoding], :or {:transfer-encoding :quoted-printable}}) fails spec: :clojure.core.specs/arg-list at: [:args :bs :arity-1 :args] predicate: (cat :args (* :clojure.core.specs/binding-form) :varargs (? (cat :amp #{(quote &)} :form :clojure.core.specs/binding-form))),  Extra input
                                         In: [2 0] val: {:keys [name content mime-type charset transfer-encoding], :or {:transfer-encoding :quoted-printable}} fails spec: :clojure.core.specs/arg-list at: [:args :bs :arity-n :bodies :args] predicate: vector?
                                         :clojure.spec/args  (populate-part "Generates a part map of the appropriate format" [{:keys [name content mime-type charset transfer-encoding], :or {:transfer-encoding :quoted-printable}}] (let [mt (or mime-type (when (instance? File content) (URLConnection/guessContentTypeFromName (.getName content))))] {:name name, :content (bs/to-byte-buffer content), :mime-type (mime-type-descriptor mt charset), :transfer-encoding transfer-encoding}))
```

**Comment by ztellman** [comment]
These both seem to be due to functions copied from clj-http's middleware.  I'll take a look.

**Comment by anderseriksson** [comment]
I run into this error when upgrading a project from 1.8.0 to 1.9.0-alpha14. If I just try to update alelph to 0.4.2-alpha8 I run into another spec error in ring.swagger.swagger2-schema but that is another story... I guess...

**Comment by camdez** [comment]
Looks like @puredanger addressed this in 7d6f2f5bf743301c5a4ab0705725bf9b50b91896 back in July.

**Comment by ztellman** [comment]
Yes, thanks for reminding me to close this.
